service/discovery/openstack/account_discover.go

```diff
@@ -26,24 +26,63 @@
 package openstack
 
 import (
-	"clouditor.io/clouditor/v2/api/ontology"
+	"fmt"
 
+	"clouditor.io/clouditor/v2/api/discovery"
+	"clouditor.io/clouditor/v2/api/ontology"
 	"github.com/gophercloud/gophercloud/v2/openstack/identity/v3/domains"
 	"github.com/gophercloud/gophercloud/v2/openstack/identity/v3/projects"
 )
 
+// discoverDomains discovers domains.
+func (d *openstackDiscovery) discoverDomains() (list []ontology.IsResource, err error) {
+	var opts domains.ListOptsBuilder = &domains.ListOpts{}
+	list, err = genericList(d, d.identityClient, domains.List, d.handleDomain, domains.ExtractDomains, opts)
+
+	if err != nil {
+		// if we cannot retrieve the domain information by calling the API or from the environment variables, we will add the information manually if we already got the domain ID and/or domain name
+		log.Debugf("could not discover domains due to insufficient permissions, but we can proceed with less domain information: %v", err)
+
+		if d.domain.domainID == "" && d.domain.domainName == "" {
```

Comment:
Just a note: This reads like there could be a domainName defined without an ID - which seems weird ^^

Reply:
Yes, it is possible since you can add the domainName and domainID via the env variables. 

But maybe it is better if I only check for the ID.

Reply:
Ok, I am not sure thats why I asked. Maybe one can derive the ID via the domainName. I guess now you have to adapt the error msg as well (removing the domainName part)

Reply:
No, it is not possible to dervice the ID via the domainName. 

The error message is fixed.